pkg/handler/v1/project/project.go

```diff
@@ -48,27 +48,60 @@ func CreateEndpoint(projectProvider provider.ProjectProvider) endpoint.Endpoint
 		}
 
 		user := ctx.Value(middleware.UserCRContextKey).(*kubermaticapiv1.User)
+		settings, err := settingsProvider.GetGlobalSettings()
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		if !user.Spec.IsAdmin && settings.Spec.UserProjectsLimit != 0 {
+			userMappings, err := memberMapper.MappingsFor(user.Spec.Email)
+			if err != nil {
+				return nil, common.KubernetesErrorToHTTPError(err)
+			}
+			var errorList []string
+			var projectsCounter int64
+			for _, mapping := range userMappings {
+				userInfo := &provider.UserInfo{Email: mapping.Spec.UserEmail, Group: mapping.Spec.Group}
+				_, err := projectProvider.Get(userInfo, mapping.Spec.ProjectID, &provider.ProjectGetOptions{IncludeUninitialized: true})
+				if err != nil {
+					// Request came from the specified user. Instead `Not found` error status the `Forbidden` is returned.
+					// Next request with privileged user checks if the project doesn't exist or some other error occurred.
+					if !isStatus(err, http.StatusForbidden) {
+						errorList = append(errorList, err.Error())
+						continue
+					}
+					_, errGetUnsecured := privilegedProjectProvider.GetUnsecured(mapping.Spec.ProjectID, &provider.ProjectGetOptions{IncludeUninitialized: true})
+					if !isStatus(errGetUnsecured, http.StatusNotFound) {
+						// store original error
+						errorList = append(errorList, err.Error())
+					}
+					continue
+				}
+				projectsCounter++
```

Comment:
Does this list only owned projects? Is the restriction only for owned projects?

Reply:
all assigned projects, also with editor and viewer rights